src/metabase/upload.clj

```diff
@@ -625,7 +638,7 @@
         (driver/add-columns! driver
                              (:id database)
                              (table-identifier table)
-                             {(keyword auto-pk-column-name) (driver/upload-type->database-type driver ::auto-incrementing-int-pk)}))
+                             {auto-pk-column-keyword (conj (driver/upload-type->database-type driver ::auto-incrementing-int-pk) :primary-key)}))
```

Comment:
So the `add-columns` method is keeping `:primary-key` as part of the database type? I'm a bit disturbed by this inconsistency between the different driver public methods.

While I understand the need for an extra parameter to support a compound key, e.g. to give their order, which affects performance, I don't see the harm in marking the corresponding types with this keyword as well. 

Are you planning to change this interface as well in another PR? I guess that's fine as I don't see anything but `upload.clj` using it.

If we are going to have slightly different formats and semantics for column types between `create-table!` and `add-columns!`, it feels like quite a smell to do it as an implementation detail of this method, and extracting a first class "uptrade-type" method would lesson the evil.

Reply:
Fair point, the difference does feel a little strange. I've changed `add-column!` to be consistent cb71757bcaa331829ce68c8cb1095fee8f758b8a